### PR TITLE
sql[-parser]: remove STATISTICS INTERVAL MS Kafka option

### LINKED ISSUE #16050: rdkafka statistics.interval.ms is set to 1000ms, causing a gigantic INFO-level message to be dumped to the log every second, per sink
### What version of Materialize are you using?

08ee87e43cb5b7ac25e7ab0b968b5fac8d4e3cc4

### How did you install Materialize?

Docker image

### What is the issue?

This change:

```
commit 08ee87e43cb5b7ac25e7ab0b968b5fac8d4e3cc4
Author: Sean Loiselle <sean@materialize.io>
Date:   Fri Aug 19 17:35:25 2022 -0400

    sql: move common Kafka config option next to CONNECTION reference
```

Introduced the following default:

```
+    (StatisticsIntervalMs, i32, Default(1_000)),
```

That does not seem to have been there before. It causes a gigantic `INFO rdkafka::client: Client stats: Statistics` message to be printed every second for every sink in the system, possibly for sources as well?

I think once per second is too frequent. It needs to be much less frequent than that *and* the default handler should be overriden to not dump that `INFO` message in the log.

From the rdkafka source:

```
    /// Receives the decoded statistics of the librdkafka client. To enable, the
    /// `statistics.interval.ms` configuration parameter must be specified.
    ///
    /// The default implementation logs the statistics at the `info` log level.
    fn stats(&self, statistics: Statistics) {
        info!("Client stats: {:?}", statistics);
    }
```

@sploiselle FYI

### Relevant log output

```source
storage-u42: 2022-11-14T11:53:24.325701Z  INFO rdkafka::client: Client stats: Statistics { name: "rdkafka#producer-2", client_id: "rdkafka", client_type: "producer", ts: 754679552, time: 1668426804, age: 44005755, replyq: 0, msg_cnt: 0, msg_size: 0, msg_max: 10000000, msg_size_max: 17179869184, tx: 116, tx_bytes: 12006, rx: 116, rx_bytes: 5903, txmsgs: 38, txmsg_bytes: 1275, rxmsgs: 0, rxmsg_bytes: 0, simple_cnt: 0, metadata_cache_cnt: 2, brokers: {"TxnCoordinator": Broker { name: "TxnCoordinator", nodeid: 1, nodename: "kafka:9092", source: "logical", state: "UP", stateage: 40184548, outbuf_cnt: 0, outbuf_msg_cnt: 0, waitresp_cnt: 0, waitresp_msg_cnt: 0, tx: 72, txbytes: 4683, txerrs: 0, txretries: 0, txidle: 7818, req_timeouts: 0, rx: 72, rxbytes: 2577, rxerrs: 0, rxcorriderrs: 0, rxpartial: 0, rxidle: 7569, req: {"SaslAuthenticate": 0, "Metadata": 1, "OffsetDeleteRequest": 0, "InitProducerId": 1, "DescribeUserScramCredentialsRequest": 0, "AddPartitionsToTxn": 38, "AddOffsetsToTxn": 0, "AlterClientQuotasRequest": 0, "ApiVersion": 1, "TxnOffsetCommit": 0, "DescribeClientQuotasRequest": 0, "EndTxn": 31, "FindCoordinator": 0, "ListOffsets": 0, "SaslHandshake": 0, "Produce": 0}, zbuf_grow: 0, buf_grow: 0, wakeups: Some(389), connects: Some(6), disconnects: Some(0), int_latency: Some(Window { min: 0, max: 0, avg: 0, sum: 0, cnt: 0, stddev: 0, hdrsize: 11376, p50: 0, p75: 0, p90: 0, p95: 0, p99: 0, p99_99: 0, outofrange: 0 }), outbuf_latency: Some(Window { min: 32, max: 35, avg: 33, sum: 67, cnt: 2, stddev: 1, hdrsize: 11376, p50: 32, p75: 35, p90: 35, p95: 35, p99: 35, p99_99: 35, outofrange: 0 }), rtt: Some(Window { min: 249, max: 300, avg: 274, sum: 549, cnt: 2, stddev: 26, hdrsize: 13424, p50: 249, p75: 301, p90: 301, p95: 301, p99: 301, p99_99: 301, outofrange: 0 }), throttle: Some(Window { min: 0, max: 0, avg: 0, sum: 0, cnt: 0, stddev: 0, hdrsize: 17520, p50: 0, p75: 0, p90: 0, p95: 0, p99: 0, p99_99: 0, outofrange: 0 }), toppars: {} }, "kafka:9092/1": Broker { name: "kafka:9092/1", nodeid: 1, nodename: "kafka:9092", source: "configured", state: "UP", stateage: 44001048, outbuf_cnt: 0, outbuf_msg_cnt: 0, waitresp_cnt: 0, waitresp_msg_cnt: 0, tx: 44, txbytes: 7323, txerrs: 0, txretries: 0, txidle: 16638, req_timeouts: 0, rx: 44, rxbytes: 3326, rxerrs: 0, rxcorriderrs: 0, rxpartial: 0, rxidle: 16430, req: {"SaslHandshake": 0, "FindCoordinator": 2, "ApiVersion": 1, "AlterClientQuotasRequest": 0, "AddPartitionsToTxn": 0, "TxnOffsetCommit": 0, "OffsetDeleteRequest": 0, "AddOffsetsToTxn": 0, "Metadata": 3, "DescribeUserScramCredentialsRequest": 0, "Produce": 38, "ListOffsets": 0, "SaslAuthenticate": 0, "InitProducerId": 0, "EndTxn": 0, "DescribeClientQuotasRequest": 0}, zbuf_grow: 0, buf_grow: 0, wakeups: Some(305), connects: Some(1), disconnects: Some(0), int_latency: Some(Window { min: 1438, max: 1438, avg: 1438, sum: 1438, cnt: 1, stddev: 0, hdrsize: 14448, p50: 1439, p75: 1439, p90: 1439, p95: 1439, p99: 1439, p99_99: 1439, outofrange: 0 }), outbuf_latency: Some(Window { min: 36, max: 36, avg: 36, sum: 36, cnt: 1, stddev: 0, hdrsize: 11376, p50: 36, p75: 36, p90: 36, p95: 36, p99: 36, p99_99: 36, outofrange: 0 }), rtt: Some(Window { min: 208, max: 208, avg: 208, sum: 208, cnt: 1, stddev: 0, hdrsize: 13424, p50: 208, p75: 208, p90: 208, p95: 208, p99: 208, p99_99: 208, outofrange: 0 }), throttle: Some(Window { min: 0, max: 0, avg: 0, sum: 0, cnt: 0, stddev: 0, hdrsize: 17520, p50: 0, p75: 0, p90: 0, p95: 0, p99: 0, p99_99: 0, outofrange: 0 }), toppars: {"sink-sink_36-0": TopicPartition { topic: "sink-sink_36", partition: 0 }, "zippy-sink_36-1-0": TopicPartition { topic: "zippy-sink_36-1", partition: 0 }} }}, topics: {"sink-sink_36": Topic { topic: "sink-sink_36", metadata_age: 12004, batchsize: Window { min: 0, max: 0, avg: 0, sum: 0, cnt: 0, stddev: 0, hdrsize: 14448, p50: 0, p75: 0, p90: 0, p95: 0, p99: 0, p99_99: 0, outofrange: 0 }, batchcnt: Window { min: 0, max: 0, avg: 0, sum: 0, cnt: 0, stddev: 0, hdrsize: 8304, p50: 0, p75: 0, p90: 0, p95: 0, p99: 0, p99_99: 0, outofrange: 0 }, partitions: {-1: Partition { partition: -1, broker: -1, leader: -1, desired: false, unknown: false, msgq_cnt: 0, msgq_bytes: 0, xmit_msgq_cnt: 0, xmit_msgq_bytes: 0, fetchq_cnt: 0, fetchq_size: 0, fetch_state: "none", query_offset: -1001, next_offset: 0, app_offset: -1001, stored_offset: -1001, committed_offset: -1001, eof_offset: -1001, lo_offset: -1001, hi_offset: -1001, ls_offset: -1001, consumer_lag: -1, consumer_lag_stored: -1, txmsgs: 0, txbytes: 0, rxmsgs: 0, rxbytes: 0, msgs: 1, rx_ver_drops: 0, msgs_inflight: 0, next_ack_seq: 0, next_err_seq: 0, acked_msgid: 0 }, 0: Partition { partition: 0, broker: 1, leader: 1, desired: false, unknown: false, msgq_cnt: 0, msgq_bytes: 0, xmit_msgq_cnt: 0, xmit_msgq_bytes: 0, fetchq_cnt: 0, fetchq_size: 0, fetch_state: "none", query_offset: -1001, next_offset: 0, app_offset: -1001, stored_offset: -1001, committed_offset: -1001, eof_offset: -1001, lo_offset: -1001, hi_offset: -1001, ls_offset: -1001, consumer_lag: -1, consumer_lag_stored: -1, txmsgs: 7, txbytes: 97, rxmsgs: 0, rxbytes: 0, msgs: 7, rx_ver_drops: 0, msgs_inflight: 0, next_ack_seq: 7, next_err_seq: 7, acked_msgid: 7 }} }, "zippy-sink_36-1": Topic { topic: "zippy-sink_36-1", metadata_age: 24004, batchsize: Window { min: 106, max: 106, avg: 106, sum: 106, cnt: 1, stddev: 0, hdrsize: 14448, p50: 106, p75: 106, p90: 106, p95: 106, p99: 106, p99_99: 106, outofrange: 0 }, batchcnt: Window { min: 1, max: 1, avg: 1, sum: 1, cnt: 1, stddev: 0, hdrsize: 8304, p50: 1, p75: 1, p90: 1, p95: 1, p99: 1, p99_99: 1, outofrange: 0 }, partitions: {0: Partition { partition: 0, broker: 1, leader: 1, desired: false, unknown: false, msgq_cnt: 0, msgq_bytes: 0, xmit_msgq_cnt: 0, xmit_msgq_bytes: 0, fetchq_cnt: 0, fetchq_size: 0, fetch_state: "none", query_offset: -1001, next_offset: 0, app_offset: -1001, stored_offset: -1001, committed_offset: -1001, eof_offset: -1001, lo_offset: -1001, hi_offset: -1001, ls_offset: -1001, consumer_lag: -1, consumer_lag_stored: -1, txmsgs: 31, txbytes: 1178, rxmsgs: 0, rxbytes: 0, msgs: 31, rx_ver_drops: 0, msgs_inflight: 0, next_ack_seq: 31, next_err_seq: 31, acked_msgid: 31 }, -1: Partition { partition: -1, broker: -1, leader: -1, desired: false, unknown: false, msgq_cnt: 0, msgq_bytes: 0, xmit_msgq_cnt: 0, xmit_msgq_bytes: 0, fetchq_cnt: 0, fetchq_size: 0, fetch_state: "none", query_offset: -1001, next_offset: 0, app_offset: -1001, stored_offset: -1001, committed_offset: -1001, eof_offset: -1001, lo_offset: -1001, hi_offset: -1001, ls_offset: -1001, consumer_lag: -1, consumer_lag_stored: -1, txmsgs: 0, txbytes: 0, rxmsgs: 0, rxbytes: 0, msgs: 1, rx_ver_drops: 0, msgs_inflight: 0, next_ack_seq: 0, next_err_seq: 0, acked_msgid: 0 }} }}, cgrp: None, eos: Some(ExactlyOnceSemantics { idemp_state: "Assigned", idemp_stateage: 40183, txn_state: "Ready", txn_stateage: 754679, txn_may_enq: false, producer_id: 3, producer_epoch: 6, epoch_cnt: 1 }) }
```

### PR DESCRIPTION
**Pull request by benesch**
The STATISTICS INTERVAL MS option controls how often librdkafka statistics are dumped to stderr via an `info!` log. This is no longer a useful option to expose to end users, as they do not have access to Materialize's stderr logs. So just remove it.

In the future, we may want to force-enable the statistics callback with something smarter, that extracts relevant stats in librdkafka as Prometheus metrics in Materialize, but that's a project for another day.

Fix #16050.

### Motivation

  * This PR fixes a recognized bug.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: src/sql-parser/src/parser.rs [hunk 1]
```diff
@@ -2181,10 +2181,6 @@ impl<'a> Parser<'a> {
                 MS => KafkaConfigOptionName::RetentionMs,
                 _ => unreachable!(),
             },
-            STATISTICS => {
```
Comment (aljoscha) [inline src/sql-parser/src/parser.rs:2184]: the `STATISTICS` a couple lines up also needs to be removed

**Review by aljoscha — APPROVED** [review]
Looks good! I commented on the likely culprit for the build failure

**Comment by philip-stoev** [comment]
This one is blocking the installation of a Release Qualification Test in the CI. The logs from the materialized container are so large that they can not be exfiltrated out of Buildkite.

**Comment by aljoscha** [comment]
@philip-stoev I pushed a commit that should hopefully fix this PR. Should I merge once it's green or do you want to verify that it actually fixes your problem?

**Comment by philip-stoev** [comment]
Please merge on green, thanks.

**Comment by benesch** [comment]
Thanks for pushing this one through, @aljoscha!

**Pull request merged**